src/transeq.f90

```diff
@@ -737,7 +737,11 @@ subroutine momentum_gravity(dux1, duy1, duz1, peculiar_density1, richardson)
     integer :: iend, jend, kend
     integer :: i, j, k
 
+    ! Return directly if richardson is zero
+    if (abs(richardson) < epsilon(richardson)) return
```

Comment:
Should be `tiny` rather than `epsilon` to be "the smallest possible non-zero number"